app/src/main/java/org/schabi/newpipe/util/NavigationHelper.java

```diff
@@ -217,40 +220,94 @@ public static void enqueueNextOnPlayer(final Context context, final PlayQueue qu
     }
 
-    public static void playOnExternalVideoPlayer(@NonNull final Context context,
+    public static void playOnExternalVideoPlayer(final Context context,
                                                  @NonNull final StreamInfo info) {
-        final ArrayList<VideoStream> videoStreamsList = new ArrayList<>(
-                ListHelper.getSortedStreamVideosList(context, info.getVideoStreams(), null, false,
-                        false));
-        final int index = ListHelper.getDefaultResolutionIndex(context, videoStreamsList);
-
-        if (index == -1) {
+        final List<VideoStream> videoStreams = info.getVideoStreams();
+        if (videoStreams.isEmpty()) {
             Toast.makeText(context, R.string.video_streams_empty, Toast.LENGTH_SHORT).show();
             return;
         }
+        final List<VideoStream> videoStreamsForExternalPlayers =
+                ListHelper.getSortedStreamVideosList(context,
+                        removeNonUrlAndTorrentStreams(videoStreams), null, false, false);
+        if (videoStreamsForExternalPlayers.isEmpty()) {
+            Toast.makeText(context, R.string.no_video_streams_available_for_external_players,
+                    Toast.LENGTH_SHORT).show();
+            return;
+        }
+        final int index = ListHelper.getDefaultResolutionIndex(context,
+                videoStreamsForExternalPlayers);
 
-        final VideoStream videoStream = videoStreamsList.get(index);
+        final VideoStream videoStream = videoStreamsForExternalPlayers.get(index);
         playOnExternalPlayer(context, info.getName(), info.getUploaderName(), videoStream);
     }
 
     public static void playOnExternalPlayer(@NonNull final Context context,
                                             @Nullable final String name,
                                             @Nullable final String artist,
                                             @NonNull final Stream stream) {
+        final DeliveryMethod deliveryMethod = stream.getDeliveryMethod();
+        final String mimeType;
+        if (deliveryMethod == DeliveryMethod.PROGRESSIVE_HTTP) {
+            if (stream.getFormat() != null) {
+                mimeType = stream.getFormat().getMimeType();
+            } else {
+                if (stream instanceof AudioStream) {
+                    mimeType = "audio/*";
+                } else if (stream instanceof VideoStream) {
+                    mimeType = "video/*";
+                } else {
+                    // This should never be reached, because subtitles are not opened in external
+                    // players
+                    return;
+                }
+            }
+        } else {
+            if (!stream.isUrl() || deliveryMethod == DeliveryMethod.TORRENT) {
+                Toast.makeText(context, R.string.selected_stream_external_player_not_supported,
+                        Toast.LENGTH_SHORT).show();
```

Comment:
Uhm, this looks strange. Isn't `playOnExternalPlayer` always called with a valid stream?